staging/src/k8s.io/apiserver/pkg/authorization/path/path.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package path
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+// NewAuthorizer returns an authorizer which accepts a given set of paths.
+// Each path is either a fully matching path or it ends in * in case a prefix match is done. A leading / is optional.
+func NewAuthorizer(allowedPaths []string) authorizer.Authorizer {
+	var prefixes []string
+	paths := sets.NewString()
+	for _, p := range allowedPaths {
+		p = strings.TrimLeft(p, "/")
+		if strings.HasSuffix(p, "*") {
+			prefixes = append(prefixes, p[:len(p)-1])
+		} else {
+			paths.Insert(p)
+		}
+	}
+	return authorizer.AuthorizerFunc(func(a authorizer.Attributes) (authorizer.Decision, string, error) {
```

Comment:
should this only authorize non-resource requests? e.g. not authorize `/api/v1/namespaces/foo`?

Reply:
good question. I have no strong opinion.

Reply:
yes, anything doing authorization based on the authorizer.Attributes path should limit itself to non-resource requests

Reply:
done